src/protocol.rs

```diff
@@ -394,8 +398,46 @@ impl std::fmt::Display for ParticipantIdentifier {
     }
 }
 
-/// A generic identifier
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+
+/// An [`Identifier`] is a session identifier that uniquely identifies a single
+/// instance of a protocol and all messages associated with it.
+///
+/// [`Identifier`] are globally unique identifiers that allow parties to
+/// distinguish between different sessions. The "globally unique" property
+/// of these identifiers forces parties to avoid any collisions between
+/// different sessions and any kind of replay attack by associating messages,
+/// parameters, proofs and commitments with their corresponding sessions.
+///
+/// Discrepancies with the paper with respect to session identifiers:
+///
+/// Discrepancy (A): The paper distinguishes between the Session identifiers
+/// (sid) which are created from shared parameters in the protocol at the
+/// beginning of key generation and the Sub-Session identifiers (ssid) that are
+/// created from sid and other post key generation shared parameters for the
+/// purpose of pre/signing. The codebase does not make this kind of distinction
+/// relying instead on a single type for all session identifiers. The
+/// distinction between sessions and sub-sessions is inherently enforced by the
+/// order of inputs and outputs to different stages in the protocol.
+///
+/// Discrepancy (B): The codebase  instantiates [`Identifier`]s in four
+/// different ways: (1) `keygen_identifier`s that are created at  the start of a
+/// key generation instance; (2) `keyshare_identifier`s that are created after
+/// key generation; and uniquely identify a public key share (3)
+/// `presign_identifier`s that are created during the pre-signing/signing
+/// stages; (4) `auxinfo_identifier`s that are specifically created for
+/// auxiliary information generation sessions; The paper itself only
+/// distinguishes between sessions combining (1) and (2), and sub-session that
+/// combine (3) and (4).
+///
+/// Discrepancy (C): In the paper ssid is periodically refreshed with the
+/// session key and auxiliary information. The codebase does not do that and
+/// instead constructs these identifiers by randomly sampling 32 bytes. This
```

Comment:
suggestion: We might clarify here that the calling application has to handle session ID refreshing and is expected to choose a new, random `Identifier` (via the `random()` method) for every new protocol run.